src/core/Database.cpp

```diff
@@ -355,3 +320,34 @@ const CompositeKey & Database::key() const
     return m_data.key;
 }
 
+QVariantMap& Database::kdfParams() {
+    return m_data.kdfParams;
+}
+
+void Database::setKdfParams(QVariantMap params) {
```

Comment:
Yes, the docs say that, but we are not only using QtCore classes. If you get the habit of passing classes by value, you may also do that for other classes which don't implement implicit sharing. Just using const references everywhere makes things consistent and you don't need to check for every class if it does implicit sharing or not.

> Let's not talk about if the documentation was incorrect, because that's another problem.

I don't say they are wrong. I only say that I like const references better for the reasons explained above. And I would assume they are also slightly faster.